python/pyarrow/tests/test_substrait.py

```diff
@@ -21,8 +21,10 @@
 import pytest
 
 import pyarrow as pa
+import pyarrow.compute as pc
+import pyarrow.dataset as ds
```

Comment:
@westonpace It seems that we can't import `pyarrow.dataset` unconditionally: https://github.com/ursacomputing/crossbow/actions/runs/5957902192/job/16161350103#step:3:8002

```text
==================================== ERRORS ====================================
_______________ ERROR collecting pyarrow/tests/test_substrait.py _______________
ImportError while importing test module '/arrow/python/pyarrow/tests/test_substrait.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/miniconda-for-arrow/envs/pyarrow-3.10/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
pyarrow/tests/test_substrait.py:25: in <module>
    import pyarrow.dataset as ds
pyarrow/dataset.py:58: in <module>
    raise ImportError(
E   ImportError: The pyarrow installation is not built with support for 'dataset' (No module named 'pyarrow._dataset')
```

Reply:
Thanks!